### DISCUSSION TITLE
Is it possible to control the speed of speech in MixerTTS?

### DISCUSSION THREAD

**Discussion by godspirit00** [NVIDIA-NeMo/NeMo#5229] [Q&A] [answered]
I see in the description that MixerTTS is similar to FastPitch, but I don't see an argument for controlling the speed of speech in MixerTTS's `generate_spectrogram()` as the `pace` argument in FastPitch's.
So, is it possible to control the speed of speech in MixerTTS? 
Thanks!

**Comment by redoctopus** [comment, marked as answer]
We don't support this feature as an argument to Mixer-TTS by default, unfortunately. However, you might be able to hack in the `pace` argument by reaching in and adding it to the `regulate_len()` call in Mixer-TTS's `forward()` function [here](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/models/mixer_tts.py#L332).

**Reply by godspirit00** [reply]
Before posting the question I tried adding `pace` in the `regulate_len()` call in `infer()` at both L398 and L402 as I see that `generate_spectrogram()` calls `infer()` at L653. But the result is the same.
I will try adding it to `forward()` instead.

**Reply by godspirit00** [reply]
And the result still remained the same after adding `pace` to the `regulate_len()` in `infer()` and `forward()`, and finally `generate_spectrogram()`.

**Reply by redoctopus** [reply]
Oh, you're right--the `pace` argument should be inserted at L402 if you're calling `generate_spectrogram()`.

I am able to get it to work--I tried `pace=0.5` and got audio with a noticeable slowdown. Can you double-check the result? If you are using a Colab notebook, you may have to restart the kernel if you have changed the code.

**Reply by godspirit00** [reply]
Oh, I realize I also need to add the argument into `input_types` dict. After the addition, the speech slowed down as expected. Thanks for the help!
